docs/integrations/flink.md

```diff
@@ -0,0 +1,95 @@
+Apache Flink has multiple deployment modes - Session Mode, Application Mode and Per-Job mode. The most popular
+are Session Mode and Application Mode. Session Mode consists of a `JobManager` managing multiple jobs sharing single
+Flink cluster. In this mode, `JobClient` is executed on a machine that submits the job to the cluster.
+
+Application Mode is used where cluster is utilized for a single job. In this mode, `JobClient`, where the main method runs,
+is executed on the `JobManager`.
+
+Flink jobs read data from `Sources` and write data to `Sinks`. In contrast to systems like Apache Spark, Flink jobs can write 
+data to multiple places - they can have multiple `Sinks`.
+
+## Getting lineage from Flink
+
+OpenLineage utilizes Flink's `JobListener` interface. This interface is used by Flink to notify user of job submission,
+successful finish of job, or job failure. Implementations of this interface are executed on `JobClient`. 
+
+When OpenLineage listener receives information that job was submitted, it extracts `Transformations` from job's 
+`ExecutionEnvironment`. The `Transformations` represent logical operations in the dataflow graph; they are composed
+of both Flink's build-in operators, but also user-provided `Sources`, `Sinks` and functions. To get the lineage,
+OpenLineage integration processes dataflow graph. Currently, OpenLineage is interested only in information contained 
+in `Sources` and `Sinks`, as they are the places where Flink interacts with external systems. 
+
+After job submission, OpenLineage integration starts actively listen to checkpoints - this gives insight into 
+whether the job runs properly.
+
+## Limitations
+
+Currently OpenLineage's Flink integration is limited to getting information from jobs running in Application Mode.
+
+OpenLineage integration extracts lineage only from following `Sources` and `Sinks`:
+
+<table>
+  <tbody>
+    <tr>
+      <th>Sources</th>
+      <th>Sinks</th>
+    </tr>
+    <tr>
+      <td>KafkaSource</td>
+      <td>KafkaSink</td>
+    </tr>
+    <tr>
+      <td>FlinkKafkaConsumer</td>
+      <td>FlinkKafkaProducer</td>
+    </tr>
+    <tr>
+      <td>IcebergFlinkSource</td>
+      <td></td>
+    </tr>
+  </tbody>
+</table>
+
+We expect this list to grow as we add support for more connectors.
+
+## Usage
+
+First, OpenLineage jar needs to be present on `JobManager`.
+
+Also, OpenLineage needs certain parameters to be set in `flink-conf.yaml`:
+
+<table>
+  <tbody>
+<tr>
+  <th>Configuration Key</th>
+  <th>Description</th>
+  <th>Expected Value</th>
+  <th>Default</th>
+</tr>
+<tr>
+  <td>execution.attached</td>
+  <td>This setting needs to be true if OpenLineage is to detect job start and failure</td>
+  <td>true</td>
+  <td>false</td>
+</tr>
+  </tbody>
+</table>
+
+In addition to that, you need to set up OpenLineage client: simplest way is to set up `OPENLINEAGE_URL` environment
+variable to URL of your OpenLineage consumer. More advanced settings are [here](../client/java.md).
```

Comment:
It would have been great to include some code from our integration-tests apps. We still need to create `JobListener` and register it in the environment:
`env.registerJobListener(openlineageJobListener);`

Reply:
Done.